src/web/RootView.tsx

```diff
@@ -23,11 +23,13 @@ import Styles from './Styles';
 import Types = require('../common/Types');
 import FocusManager from './utils/FocusManager';
 import UserInterface from './UserInterface';
+import PopupContainer from './PopupContainer';
 
 export interface RootViewProps {
     mainView?: React.ReactNode;
     modal?: React.ReactElement<Types.ViewProps>;
-    activePopupOptions?: Types.PopupOptions;
+    activePopupOptions?: { options: Types.PopupOptions, id: string };
```

Comment:
See comment above about using a named type here rather than duplicating the type definition.

Reply:
Also, the name "activePopupOptions" should be changed since this is no longer just the options. Perhaps "activePopup" or "activePopupInfo". Same below for "cachedPopupOptions".

Reply:
Done.